data/state/peerAccount.go

```diff
@@ -61,8 +71,13 @@ func (pa *peerAccount) SetRewardAddress(address []byte) error {
 }
 
 // SetStake sets the account's stake
-func (pa *peerAccount) SetStake(stake *big.Int) {
+func (pa *peerAccount) SetStake(stake *big.Int) error {
+	if stake == nil {
+		return ErrNilStake
+	}
+
 	pa.Stake = stake
```

Comment:
suggestion to change to
```GO
pa.Stake = big.NewInt(0).Set(stake)
```
In order to copy the value and reduce the chances that the provided pointer, if altered after this call will impact the stored value.

Reply:
Done.